django_mongodb/functions.py

```diff
@@ -62,7 +62,10 @@ def cast(self, compiler, connection):
     lhs_mql = process_lhs(self, compiler, connection)[0]
     if max_length := self.output_field.max_length:
         lhs_mql = {"$substrCP": [lhs_mql, 0, max_length]}
-    lhs_mql = {"$convert": {"input": lhs_mql, "to": output_type}}
+    if output_type == "object":
+        lhs_mql = {"$convert": {"input": lhs_mql, "to": output_type, "onError": lhs_mql}}
```

Comment:
A clarifying comment would be useful explaining in what case this fails and why returning the original value is the correct solution.